rs/nns/governance/src/neuron/types.rs

```diff
@@ -1794,5 +1795,78 @@ impl TryFrom<StoredDissolveStateAndAge> for DissolveStateAndAge {
     }
 }
 
+/// Adds up the potential voting power of all neurons.
+///
+/// # Potential Voting Power vs. Deciding Voting Power
+///
+/// If all neurons keep themselves refreshed, then deciding voting power =
+/// potential voting power. Whereas, if a neuron has not refreshed recently
+/// enough, the amount of voting power it can exercise is less than its
+/// potential.
+///
+/// # Weird Special Case: ManageNeuron
+///
+/// Voting power is weird in the case of ManageNeuron
+/// proposals. In that case, only the followees of the targetted neuron can
+/// vote, and they all get 1 voting power, regardless of refresh. Also,
+/// these proposals have no rewards. Therefore, in the case of ManageNeuron,
+/// this returns ballots_len.
+///
+/// # When Err
+///
+/// There is no "practical" input that can cause Err to be returned.
+/// (See comments in the body.)
+pub(crate) fn total_potential_voting_power<'a>(
```

Comment:
So much code to for calculating a sum...

Part of the reason for so much code is that the world (in particular, Rust and Protocol  Buffers) believes that a fixed number of bits scales, because "64 bits ought to be enough for anybody".

The other sin that leads to so much code is ManageNeuron proposals.